src/content/blog/astro-5140.mdx

```diff
@@ -0,0 +1,292 @@
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Prerendered route collision warnings
+
+Astro 5.14 makes it easier to spot and fix common errors in dynamic routing before they cause problems in production.
+
+Previously, when two dynamic routes such as `/blog/[slug]` and `/blog/[...all]` tried to prerender the same path (e.g. `/blog/post-1` ), Astro rendered only the [highest priority route](https://docs.astro.build/en/guides/routing/#route-priority-order) for the conflicting path while the other would be silently ignored. This allowed your site to build successfully, although you may have discovered that some pages were rendered by unexpected routes.
+
+Now, when this happens, a warning will be displayed explaining which routes collided and on which path.
+
+Additionally, a new experimental flag `failOnPrerenderCollision` can be used to fail the build with a helpful error when such a collision occurs. Set the following flag in your Astro config for "Houston's helping hand" to force you to explicitly address any routing conflicts:
+
+```js
+// astro.config.mjs
+export default defineConfig({
+	experimental: {
+		failOnPrerenderCollision: true,
+	},
+});
+```
+
+See more in the [experimental prerender conflict error](https://docs.astro.build/en/reference/experimental-flags/fail-on-prerender-conflict/) documentation.
+
+## `routePattern` in `getStaticPaths`
+
+Astro 5.14 makes it easier to work with complex dynamic routes in `getStaticPaths()`.
+
+A new `routePattern` property is now available in the `getStaticPaths` context. This provides the original, dynamic segment definition in a routing file path (e.g. `/[...locale]/[files]/[slug]`) from the Astro render context that would not otherwise be available within the scope of `getStaticPaths()`. This can be useful to calculate the `params` and `props` for each page route.
+
+For example, you can now localize your route segments and return an array of static paths by passing `routePattern` to a custom `getLocalizedData()` helper function. The `params` object will be set with explicit values for each route segment (e.g. `locale`, `files`, and `slug`). Then, these values will be used to generate the routes and can be used in your page template via `Astro.params`.
+
+```astro
+// src/pages/[...locale]/[files]/[slug].astro
+
+import { getLocalizedData } from "../../../utils/i18n"; export async function getStaticPaths({
+routePattern }) { const response = await fetch('...'); const data = await response.json();
+console.log(routePattern); // [...locale]/[files]/[slug] // Call your custom helper
+with `routePattern` to generate the static paths return data.flatMap((file) => getLocalizedData(file,
+routePattern)); } const { locale, files, slug } = Astro.params;
```

Comment:
```suggestion
// src/pages/[...locale]/[files]/[slug].astro

import { getLocalizedData } from '../../../utils/i18n';

export async function getStaticPaths({ routePattern }) {
	const response = await fetch('...');
	const data = await response.json();
	console.log(routePattern);
	// [...locale]/[files]/[slug]
	// Call your custom helper with `routePattern` to generate the static paths
	return data.flatMap((file) => getLocalizedData(file, routePattern));
}

const { locale, files, slug } = Astro.params;
```

I guess something went wrong with this code block. Maybe double check if the suggested formatting is the one we want.